### DISCUSSION TITLE
15s timeout even on Pro account

### DISCUSSION THREAD

**Discussion by FrancescoSaverioZuppichini** [vercel/community#4950] [Help] [answered]
### Summary

Dear all,

I am experiencing a 15s timeout with a **Pro** account and **stream**. I have an endpoint that uses `gpt-4` to stream some data, you can see the network tab

![Screenshot 2023-11-23 at 23 02 53](https://github.com/vercel/community/assets/15908060/f93b122d-18f7-4cc1-931d-0a172c590a8c)

the endpoint just starts an open ai stream request, it works locally

```javascript
 const systemPrompt = createSystemPrompt(chat)

    const response = await openai.createChatCompletion({
      // model: "gpt-3.5-turbo", 
      model: "gpt-4-1106-preview", 
      // max_tokens: 10,
      response_format: { type: "json_object" },
      stream: true,
      messages: [
        {
          role: "system",
          content: systemPrompt,
        },
      ],
    })
    const startTime = performance.now();

    const stream = OpenAIStream(response, {
      onCompletion: async (completion: string) => {
        const evaluation = await onCompletion(completion, body.chat.id);
        const endTime = performance.now();
        console.log(`Evaluation ${evaluation.id} creation tooks ${(endTime - startTime) / 1000}`);

      },
    })
    

    return new StreamingTextResponse(stream)
```

Vercel logs

![Screenshot 2023-11-23 at 23 04 06](https://github.com/vercel/community/assets/15908060/b580d7e2-7639-4b57-b5b5-2e61692b028a)

Thanks a lot

Fra

### Steps to Reproduce

Not sure how to advice

**Comment by leerob** [comment, marked as answer]
Hey! This is correct. If you want to increase the timeout, you can use the `maxDuration` configuration: https://vercel.com/docs/functions/serverless-functions/runtimes#max-duration
